nomad/state/paginator/filter.go

```diff
@@ -3,42 +3,19 @@
 
 package paginator
 
-// Filter is the interface that must be implemented to skip values when using
-// the Paginator.
-type Filter interface {
-	// Evaluate returns true if the element should be added to the page.
-	Evaluate(interface{}) (bool, error)
-}
-
-// GenericFilter wraps a function that can be used to provide simple or in
-// scope filtering.
-type GenericFilter struct {
-	Allow func(interface{}) (bool, error)
-}
-
-func (f GenericFilter) Evaluate(raw interface{}) (bool, error) {
-	return f.Allow(raw)
-}
-
-// NamespaceFilter skips elements with a namespace value that is not in the
-// allowable set.
-type NamespaceFilter struct {
-	AllowableNamespaces map[string]bool
-}
-
-func (f NamespaceFilter) Evaluate(raw interface{}) (bool, error) {
-	if raw == nil {
-		return false, nil
-	}
-
-	item, _ := raw.(NamespaceGetter)
-	namespace := item.GetNamespace()
-
-	if f.AllowableNamespaces == nil {
-		return true, nil
-	}
-	if f.AllowableNamespaces[namespace] {
-		return true, nil
+// FilterFunc is an interface for functions that return true if the object
+// evaluated should be included in the page.
+//
+// Warning: this is the opposite of a memdb.FilterFunc, where returning true
+// excludes the object!
```

Comment:
This is surely going to result in a few snafus but I wonder: is there a better name for this than "filter"?

Reply:
Good call... what about `SelectorFunc`? That way we're _selecting_ an object, rather than filtering and having to guess whether that means to select or reject.

Reply:
Oh that's a nice one, I like it.

Reply:
Done in [257792b](https://github.com/hashicorp/nomad/pull/25252/commits/257792b2bf78dae44610b8191a51f364ef688329)